src/SALib/analyze/morris.py

```diff
@@ -345,8 +392,32 @@ def _compute_elementary_effects(
         output_matrix, value_increased, value_decreased, increase=False
     )
 
-    elementary_effects = _calc_results_difference(result_increased, result_decreased)
-    np.divide(elementary_effects, delta, out=elementary_effects)
+    difference = _calc_results_difference(result_increased, result_decreased)
+
+    if scaling is True:
+
+        # calculate the scaled value of delta (step size)
+        dx_trajectory = _calculate_step_size_x(input_matrix)
+
+        see_dy_dx = np.divide(difference, dx_trajectory.T)
+
+        sigma_x_per_trajectory = np.std(input_matrix, axis=1)
+        sigma_y_per_trajectory = np.std(output_matrix, axis=1)
+
+        # Create a mask to avoid division by zero
+        mask = sigma_y_per_trajectory[:, np.newaxis] != 0
+
+        adjustment_trajectory = np.divide(
+            sigma_x_per_trajectory,
+            sigma_y_per_trajectory[:, np.newaxis],
+            out=np.zeros_like(sigma_x_per_trajectory),
+            where=mask,
+        )
+        elementary_effects = see_dy_dx * adjustment_trajectory.T
+
+    else:
+
+        elementary_effects = np.divide(difference, delta)
```

Comment:
Is use of `np.divide` required? (as opposed to `difference / delta`)

Reply:
No, I think you are correct as delta is a float in this case. This is the original formulation,

Reply:
In this case, please remove use of `np.divide`.

The `/` is shorthand for `np.divide` anyway. 
https://numpy.org/doc/stable/reference/generated/numpy.divide.html

If you're interested in the technical details, on top of the above Python has a constant function call overhead - meaning just calling a function (any function) adds $x$ amount of time. Admittedly it is a small amount of time, so it's a micro-optimisation but I usually try to err on the side of simplicity :)

![image](https://github.com/SALib/SALib/assets/4075136/f86a263e-8015-4ad0-b34d-56cd671ade04)